### PR TITLE
Feature/160663914 universal did resolver

### PR DESCRIPTION
**Pull request by meaning-matters**
Adds the `UniversalDIDResolver` that can hold multiple DID resolvers that implement the `DIDResolver` protocol.  After `DIDResolver` implementations are registered at a `UniversalDIDResolver`, it will select & use the appropriate resolver when given a DID.

At the moment only W3C formatted ("did:<method>:<id>") are supported.

Implemented are the `EthrDIDResolver` and the `UPortDIDResolver`.  The logic for both was already present in the SDK.  Adding the Ethereum `DIDResolver` implementation was fairly straightforward.  But, the uPort implementation required logical fixes to the existing resolver code.

Additionally I did code cleanup.  The major part was reformatting the SDKs Swift modules.  I've changed indentation to the Allman/BSD indentation style (https://en.wikipedia.org/wiki/Indentation_style#Allman_style), because I believe this makes, especially Swift code, easier on the eye.  Simple spacing inconsistencies were also resolved.  And, I added comment headers to a number of files.

Tests were added for all new logic.  Since the SDK's example app is still empty, the only way to tests this work is running the tests.

### CODE REVIEW

File: uPortSDK/Classes/EthrDID/EthrDIDResolver.swift [hunk 1]
```diff
@@ -9,327 +9,442 @@ import Foundation
...
+    public init(rpc: JsonRPC, registryAddress: String = DEFAULT_REGISTERY_ADDRESS)
+    {
         self.rpc = rpc
         self.registryAddress = registryAddress
     }
 
-    public func resolve( did: String ) throws -> DDO {
-        let normalizedDidObject = NormalizedDID( didCandidate: did  )
-        guard normalizedDidObject.error == nil else {
+    // MARK: - Resolver Implementation
```
[the remark this answers is not in the document]
Comment (meaning-matters) [reply uPortSDK/Classes/EthrDID/EthrDIDResolver.swift]: `// MARK: -` is used to mark a section of code.  It's also parsed by Xcode and results in placing a 'mark' (vertical line) in the top-of-screen dropdown list of functions.

File: uPortSDK/Classes/EthrDID/EthrDID.swift [hunk 2]
```diff
@@ -26,106 +28,135 @@ public struct EthrDID {
...
-                DispatchQueue.global().async {
-                    do {
-                        let response = try self.lookupOwnerSynchronous( cache: cache )
+    public func lookupOwner(cache: Bool = true) -> Promise<String>
+    {
+        return Promise<String>
+        { fulfill, reject in
+            DispatchQueue.global().async
+            {
+                DispatchQueue.global().async
```
[the remark this answers is not in the document]
Comment (meaning-matters) [reply uPortSDK/Classes/EthrDID/EthrDID.swift]: Indeed weird. That duplication was already in the original code (I just checked), which I reformatted.  I can't think of anything this construction would do, so should be safe to remove.

File: uPortSDK/Classes/EthrDID/EthrDIDResolver.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mirceanis) [reply uPortSDK/Classes/EthrDID/EthrDIDResolver.swift]: I'm learning something new every day :)

**Comment by mirceanis** [comment]
Great work, thank you!

**Pull request merged**